cdap-app-fabric/src/test/java/co/cask/cdap/scheduler/CoreSchedulerServiceTest.java

```diff
@@ -423,6 +499,14 @@ public Integer call() throws Exception {
     }, 10, TimeUnit.SECONDS);
   }
 
+  private ProgramRunId getLatestRun(ProgramId workflowId) {
+    return (ProgramRunId) getAllRuns(workflowId).toArray()[0];
```

Comment:
this doesn't look like getting last run. `store.getRuns` with `ProgramRunStatus.ALL` seems to return run records in the order of active runs -> suspended -> historical, sorted by start time. Check the order. Also why `toArray()`?